trace_decoder/tests/consistent-with-header.rs

```diff
@@ -0,0 +1,70 @@
+//! Check that the [`evm_arithmetization::GenerationInputs`] produced by
+//! [`trace_decoder`] are consistent between each other, and with the block
+//! header obtained over RPC.
+
+mod common;
+
+use alloy_compat::Compat as _;
+use assert2::check;
+use common::{cases, Case};
+use itertools::Itertools;
+use libtest_mimic::{Arguments, Trial};
+use mpt_trie::partial_trie::PartialTrie as _;
+
+fn main() -> anyhow::Result<()> {
+    let mut trials = vec![];
+    for Case {
+        name,
+        header,
+        trace,
+        other,
+    } in cases()?
+    {
+        trials.push(Trial::test(name, move || {
+            let gis = trace_decoder::entrypoint(trace.clone(), other.clone(), 3, false)
+                .map_err(|e| format!("{e:?}"))?; // get the full cause chain
+            check!(gis.len() >= 2);
+            check!(
+                Some(other.checkpoint_state_trie_root)
+                    == gis.first().map(|it| it.tries.state_trie.hash())
```

Comment:
we just enforced to have at least 2 so why not calling 
```suggestion
                other.checkpoint_state_trie_root
                    == gis[0].tries.state_trie.hash()
```

Reply:
Doing it this way means that multiple failures can occur without panicking until after they're all checked

Reply:
Ah got it, thanks.